lib/couchdb-client.go

```diff
@@ -254,69 +255,129 @@ func (c *CouchdbClient) getStats(config CollectorConfig) (Stats, error) {
 
 func (c *CouchdbClient) getDatabasesStatsByDbName(databases []string) (map[string]DatabaseStats, error) {
 	dbStatsByDbName := make(map[string]DatabaseStats)
+	type result struct {
+		dbName  string
+		dbStats DatabaseStats
+		err     error
+	}
+	r := make(chan result, len(databases))
+	// scatter
 	for _, dbName := range databases {
-		data, err := c.Request("GET", fmt.Sprintf("%s/%s", c.BaseUri, dbName), nil)
-		if err != nil {
-			return nil, fmt.Errorf("error reading database '%s' stats: %v", dbName, err)
-		}
+		dbName := dbName // rebind for closure to capture the value
+		go func() {
+			var dbStats DatabaseStats
+			data, err := c.Request("GET", fmt.Sprintf("%s/%s", c.BaseUri, dbName), nil)
+			if err != nil {
+				r <- result{err: fmt.Errorf("error reading database '%s' stats: %v", dbName, err)}
+				return
+			}
 
-		var dbStats DatabaseStats
-		err = json.Unmarshal(data, &dbStats)
-		if err != nil {
-			return nil, fmt.Errorf("error unmarshalling database '%s' stats: %v", dbName, err)
-		}
-		dbStats.DiskSizeOverhead = dbStats.DiskSize - dbStats.DataSize
-		if dbStats.CompactRunningBool {
-			dbStats.CompactRunning = 1
-		} else {
-			dbStats.CompactRunning = 0
+			err = json.Unmarshal(data, &dbStats)
+			if err != nil {
+				r <- result{err: fmt.Errorf("error unmarshalling database '%s' stats: %v", dbName, err)}
+				return
+			}
+			dbStats.DiskSizeOverhead = dbStats.DiskSize - dbStats.DataSize
+			if dbStats.CompactRunningBool {
+				dbStats.CompactRunning = 1
+			} else {
+				dbStats.CompactRunning = 0
+			}
+			r <- result{dbName, dbStats, nil}
+		}()
+	}
+	// gather
+	for _ = range databases {
```

Comment:
The `_ = ` isn't necessary - at least my IDE told me so ;)

```suggestion
	for range databases {
```